lgc/builder/SubgroupBuilder.cpp

```diff
@@ -134,8 +134,20 @@ Value *SubgroupBuilder::CreateSubgroupAllEqual(Value *const value, bool wqm, con
 //
 // @param value : The value to read from the chosen lane to all active lanes.
 // @param index : The index to broadcast from. Must be an i32.
+// @param relaxed : The index may be non-uniform (undefined). Handle with waterfall.
 // @param instName : Name to give final instruction.
-Value *SubgroupBuilder::CreateSubgroupBroadcast(Value *const value, Value *const index, const Twine &instName) {
+Value *SubgroupBuilder::CreateSubgroupBroadcast(Value *const value, Value *const index, bool relaxed,
+                                                const Twine &instName) {
```

Comment:
From a coding style perspective, I think it might be better to write a new function called "CreateSubgroupBroadcastUsingWaterfallLoop" or something like that.  The boolean creates two completely different function already.  There is no code sharing.  It also make it harder to read when most of the time you have this constant false in the function call.  It is not clear what that false mean, so you have to look up the function.

Reply:
Yes, good point. I'll update the change.
We've done some internal testing on this as well and based on that I think it's probably a reasonable and safe change.